### PR TITLE
CVSL-317: Add integration tests for AWS event consumer

### PR DESCRIPTION
**Pull request by smcveigh941**
- Created custom CircleCi executor including localstack (pathfinder and manage intelligence have done this, should add it to the orb for future and other projects?)
- Created queues in localstack for integration test
- Added new integration test helper to push events onto the stack
- Added new helper to check wiremock `count` endpoint to assert the correct endpoint is called after consuming an event
- Handled rejected promise from the event handler

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,6 +18,30 @@ parameters:
...
+      resource_class:
+        default: medium
+        type: string
+      node_tag:
+        default: "14.16-browsers"
+        type: string
+      redis_tag:
+        default: "buster"
+        type: string
+
```
Comment (timharrison-moj) [inline .circleci/config.yml:44]: Does seem like an orb candidate, as prisoner-offender-search, probation-offender-search, pathfinder, IMS and hmpps-communication-monitoring will all do this. I would have raised to dps-tech-team, but they've all moved into other teams now, so the ORB is "group-maintained". Ok for now, I'll raise in appropriate channel.

File: integration_tests/integration/eventHandlers.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      cy.task('stubUpdateLicenceStatus')
+
+      cy.task(
+        'sendDomainEvent',
+        `{
+            "Message" :  "{\\"eventType\\":\\"prison-offender-events.prisoner.released\\",\\"additionalInformation\\":{\\"nomsNumber\\":\\"A7774DY\\",\\"reason\\":\\"RELEASED\\",\\"details\\":\\"Movement reason code CR\\",\\"currentLocation\\":\\"OUTSIDE_PRISON\\",\\"prisonId\\":\\"MDI\\",\\"currentPrisonStatus\\":\\"NOT_UNDER_PRISON_CARE\\"},\\"version\\":1,\\"occurredAt\\":\\"2022-01-12T14:56:51.662128Z\\",\\"publishedAt\\":\\"2022-01-12T14:58:25.021008001Z\\",\\"description\\":\\"A prisoner has been released from prison\\"}"
+         }`
+      )
+
+      cy.task('verifyEndpointCalled', { verb: 'PUT', path: '/licence/id/1/status', times: 1 }).then(success => {
```
Comment (timharrison-moj) [inline integration_tests/integration/eventHandlers.spec.ts:20]: Any value in verifying that the GET endpoint is also called?

**Review by timharrison-moj — APPROVED** [review]
LGTM. Couple of questions only.

File: integration_tests/integration/eventHandlers.spec.ts (hunk 2 above)
Comment (smcveigh941) [reply integration_tests/integration/eventHandlers.spec.ts:20]: I dont think so, this verifies that the event is consumed. The actual handler is unit tested

**Pull request merged**